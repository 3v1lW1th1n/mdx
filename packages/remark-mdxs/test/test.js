const mdx = require('../../mdx')
const remarkMdxs = require('..')

const FIXTURE = `
import Foo from './bar'
export const author = 'fred'
export default Foo

# Hello, world! <Foo bar={{ baz: 'qux' }} />

---

<Baz>
  Hi!
</Baz>

---

# I'm another document

over here.
`

it('correctly transpiles', async () => {
  const result = await mdx(FIXTURE, {
    remarkPlugins: [remarkMdxs]
  })

  expect(result).toMatchSnapshot()
})
