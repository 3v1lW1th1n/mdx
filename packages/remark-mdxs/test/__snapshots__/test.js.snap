// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`correctly transpiles 1`] = `
"/* @jsx mdx */
import Foo from './bar'

export const author = 'fred'

const MDXSLayout = Foo

function MDXSContent0({ components, ...props }) {
  return (
    <h1>{\`Hello, world! \`}<Foo bar={{ baz: 'qux' }} /></h1>
  )
}

function MDXSContent1({ components, ...props }) {
  return (
    <Baz>
  Hi!
</Baz>
  )
}

function MDXSContent2({ components, ...props }) {
  return (
    <h1>{\`I'm another document\`}</h1>


<p>{\`over here.\`}</p>
  )
}

const MDXSWrapper = props => [
  <MDXSContent0 {...props} />,
  <MDXSContent1 {...props} />,
  <MDXSContent2 {...props} />
]

export default MDXWrapper"
`;
